src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -327,6 +290,31 @@ private async Task<long> SaveMongoUpdatesAsync(IEnumerable<MongoUpdate> updates,
         return documentsAffected;
     }
 
+    private static BsonDocument ToBsonDocument(IUpdateEntry entry, IBsonDocumentSerializer entitySerializer, IEnumerable<IPropertyBase> properties)
+    {
+        var document = new BsonDocument();
+        using (var writer = new BsonDocumentWriter(document))
+        {
+            writer.WriteStartDocument();
+
+            foreach (var property in properties)
+            {
+                var propertyValue = entry.GetCurrentValue(property);
+                var propertySerializationInfo = GetPropertySerializationInfo(entitySerializer, property);
+                var elementName = propertySerializationInfo.ElementName;
+                var propertySerializer = propertySerializationInfo.Serializer;
+                var context = BsonSerializationContext.CreateRoot(writer);
+
+                writer.WriteName(elementName);
+                propertySerializer.Serialize(context, propertyValue);
+            }
+
+            writer.WriteEndDocument();
+        }
+
+        return document;
+    }
+
```

Comment:
The above changes depend on the following new helper methods:

    private static BsonValue SerializePrimaryKey(IUpdateEntry entry, IBsonDocumentSerializer entitySerializer)
    {
        var document = new BsonDocument();
        using (var writer = new BsonDocumentWriter(document))
        {
            writer.WriteStartDocument();
            SerializePrimaryKey(writer, entry, entitySerializer);
            writer.WriteEndDocument();
        }
        return document["_id"];
    }

    private static void SerializePrimaryKey(IBsonWriter writer, IUpdateEntry entry, IBsonDocumentSerializer entitySerializer)
    {
        var primaryKey = entry.EntityType.FindPrimaryKey();
        if (primaryKey == null)
        {
            throw new InvalidOperationException("No primary key found.");
        }

        writer.WriteName("_id");
        var primaryKeyProperties = primaryKey.Properties;
        if (primaryKeyProperties.Count == 1)
        {
            SerializeProperties(writer, entry, entitySerializer, primaryKeyProperties);
        }
        else
        {

            writer.WriteStartDocument();
            SerializeProperties(writer, entry, entitySerializer, primaryKeyProperties);
            writer.WriteEndDocument();
        }
    }